include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h

```diff
@@ -95,18 +96,52 @@ exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
 }
 
 #if _ONEDPL_BACKEND_SYCL
+
 template <typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator, typename T,
           typename BinaryPredicate, typename Operator>
 oneapi::dpl::__internal::__enable_if_hetero_execution_policy<typename ::std::decay<Policy>::type, OutputIterator>
-exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIterator1 last1, InputIterator2 first2,
-                               OutputIterator result, T init, BinaryPredicate binary_pred, Operator binary_op)
+exclusive_scan_by_segment_impl_helper(Policy&& policy, InputIterator1 first1, InputIterator1 last1,
+                                      InputIterator2 first2, OutputIterator result, T init, BinaryPredicate binary_pred,
+                                      Operator binary_op, ::std::true_type /* has_known_identity*/)
 {
     const auto n = ::std::distance(first1, last1);
 
     // Check for empty element ranges
     if (n <= 0)
         return result;
 
+    typedef uint64_t CountType;
```

Comment:
Use `::std::uint64_t` instead of `uint64_t`. Also, I would prefer `using` over `typedef`, but perhaps it's a personal choice.

Reply:
I will make the change to `::std::uint64_t`. I normally use `using` as well, but used `typedef` to be consistent with the style of the already existing code in this header.